loafwallet/MainViewController.swift

```diff
@@ -91,40 +91,61 @@ class MainViewController : UIViewController, Subscriber, LoginViewControllerDele
     }
    
     func didUnlockLogin() {
```

Comment:
@kcw-grunt refactor this method like this:

```
func didUnlockLogin() {
          

var vc : UIViewController
if UserDefaults.userIsInUSA {
vc = UIStoryboard.init(name: "Main", bundle: nil)
                        .instantiateViewController(withIdentifier: "TabBarViewController")
                        as? TabBarViewController
} else {
vc = UIStoryboard.init(name: "Main", bundle: nil)
                        .instantiateViewController(withIdentifier: "NonUSTabBarViewController")
                        as? NonUSTabBarViewController
}


vc.store = self.store
vc.walletManager = walletManager
            
addChildViewController(vc, layout: {
    vc.view.constrain(toSuperviewEdges: nil)
    vc.view.alpha = 0
    vc.view.layoutIfNeeded()
})
            
UIView.animate(withDuration: 0.3,
               delay: 0.1,
               options: .transitionCrossDissolve,
               animations: {          
    vc.view.alpha = 1             
}) { (finished) in
    NSLog("\(UserDefaults.userIsInUSA ? US : Ex US)  MainView Controller presented")
}

}
```

Reply:
Oh that won't work.  The classes are custom and inherit so the values can't be passed to the NonUS and Tab ViewControllers

Reply:
hmm I see. Can you do this 👇 ?


```
func didUnlockLogin() {
          

var vc : UIViewController
if UserDefaults.userIsInUSA  {
guard let usaVC = UIStoryboard.init(name: "Main", bundle: nil)
                        .instantiateViewController(withIdentifier: "TabBarViewController")
                        as? TabBarViewController else {
                
                NSLog("TabBarViewController not intialized")
                return
            }
usaVC.store = self.store
usaVC.walletManager = walletManager
vc = usaVC
}  else  {
guard let exUSAVC = UIStoryboard.init(name: "Main", bundle: nil)
                        .instantiateViewController(withIdentifier: "NonUSTabBarViewController")
                        as? NonUSTabBarViewController else {
                
                NSLog("NonUSTabBarViewController not intialized")
                return
            }
exUSAVC.store = self.store
exUSAVC.walletManager = walletManager
vc = exUSAVC
}


addChildViewController(vc, layout: {
    vc.view.constrain(toSuperviewEdges: nil)
    vc.view.alpha = 0
    vc.view.layoutIfNeeded()
})
            
UIView.animate(withDuration: 0.3,
               delay: 0.1,
               options: .transitionCrossDissolve,
               animations: {          
    vc.view.alpha = 1             
}) { (finished) in
    NSLog("\(UserDefaults.userIsInUSA ? US : Ex US)  MainView Controller presented")
}

}
```